tools/src/tester/ResultsDisplayer.ts

```diff
@@ -0,0 +1,110 @@
+import { type ChapterEvaluation, type Evaluation, Result, type StoryEvaluation } from './types/eval.types'
+import { overall_result } from './helpers'
+
+function b (text: string): string { return `\x1b[1m${text}\x1b[0m` }
+function i (text: string): string { return `\x1b[3m${text}\x1b[0m` }
+
+function padding (text: string, length: number, prefix: number = 0): string {
+  const spaces = length - text.length > 0 ? ' '.repeat(length - text.length) : ''
+  return `${' '.repeat(prefix)}${text}${spaces}`
+}
+
+function green (text: string): string { return `\x1b[32m${text}\x1b[0m` }
+function red (text: string): string { return `\x1b[31m${text}\x1b[0m` }
+function yellow (text: string): string { return `\x1b[33m${text}\x1b[0m` }
+function cyan (text: string): string { return `\x1b[36m${text}\x1b[0m` }
+function gray (text: string): string { return `\x1b[90m${text}\x1b[0m` }
+function magenta (text: string): string { return `\x1b[35m${text}\x1b[0m` }
+
+export default class ResultsDisplayer {
+  gap: number
+  evaluation: StoryEvaluation
+  skip_components: boolean
+  ignore_passed: boolean = true
+  ignore_skipped: boolean = false
+
+  constructor (evaluation: StoryEvaluation, gap: number = 4) {
+    this.evaluation = evaluation
+    this.skip_components = [Result.PASSED, Result.SKIPPED].includes(evaluation.result)
+    this.gap = gap
+  }
+
+  display (ignore_skipped = false, ignore_passed = true): void {
+    this.ignore_passed = ignore_passed
+    this.ignore_skipped = ignore_skipped
```

Comment:
Given the constructor and display are called next to each other, these should be passed to the constructor if they're going to be assigned to member fields.